### ISSUE TITLE
Screen flickering issue on ChromeOS

### ISSUE THREAD

**Issue by chibuezeanakor** [zed-industries/zed#14284]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

When opening the Zed editor (Linux version) on my Chromebook, the app starts flickering, and I don't know how to stop it. Plus, none of the buttons in the app work when this happens, so I can't go into the app to figure out a solution to this. I attached a video to show this issue in action.

[Screen recording 2024-07-11 10.07.24 PM.webm](https://github.com/user-attachments/assets/2d2b82b4-fde3-47e3-a0ce-4f8bb7c4c9b5)

### Environment

Computer make and model: Framework Laptop - Chromebook Edition
OS: ChromeOS 126
CPU: 12th Gen Intel(R) Core(TM) i5-1240P (16 threads, 3.85GHz)
RAM: 32 GB
Storage: 1TB

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by ajorg** [comment]
I see the same sort of thing on my Samsung Galaxy Chromebook, only much slower.

**Comment by JosephTLyons** [comment]
I wonder if this is the same issue as this flickering one:

- https://github.com/zed-industries/zed/issues/14101

**Comment by ajorg** [comment]
To me it looks very different from that, like it's displaying two different UIs.

**Comment by codebird** [comment]
Same issue here. Tried some stuff such as enabling gpu for crostini in chrome flags. Low density makes the window bigger but the flickering stays...

**Comment by Trombach** [comment]
I am seeing the same thing. I think it's switching between the auto update UI and the normal UI. I've opened zed with the `--foreground` option and I can see there is an error with finding a file. I'll have a look at the source and see if I can figure out what might be missing. Although I'm not entirely sure if that's the root of the problem
![image](https://github.com/user-attachments/assets/f45d985b-2920-40e6-b366-64b1a2722be8)

**Comment by codebird** [comment]
> I am seeing the same thing. I think it's switching between the auto update UI and the normal UI. I've opened zed with the `--foreground` option and I can see there is an error with finding a file. I'll have a look at the source and see if I can figure out what might be missing. Although I'm not entirely sure if that's the root of the problem ![image](https://private-user-images.githubusercontent.com/19306765/350653568-f45d985b-2920-40e6-b366-64b1a2722be8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE0NTk1OTMsIm5iZiI6MTcyMTQ1OTI5MywicGF0aCI6Ii8xOTMwNjc2NS8zNTA2NTM1NjgtZjQ1ZDk4NWItMjkyMC00MGU2LWIzNjYtNjRiMWEyNzIyYmU4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzIwVDA3MDgxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUzZTA1MzY2NDFkYTkxZjIxMWE5OTk0ZjJhNTc0MWZmMTllZGVkNDZlMGY5OGIwMDRmM2U1N2EyYzI4Y2U5ODcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.MRjgFrgkhL7V_KTK3DTW-ToF8yLndLdt62G8th9iuOk)

For me there is no error whatsoever, I tried foreground, and since the beginning I am looking at the logs, in the logs there is a warning that might have to do something with us:

`[WARN] Unable to forbid exclusive full screen`

**Comment by Trombach** [comment]
I actually just noticed that I was running 143. After updating to the latest version of zed, the error message has disappeared, but the behaviour is the same. I'm also seeing the `[WARN] Unable to forbid exclusive full screen` in the logs, but that's all.

**Comment by codebird** [comment]
Pulled latest version, now I am getting a pop up on launch, saying that the gpu is an emulated software gpu and it is unsupported :(

**Comment by AdjectiveAllison** [comment]
Hey, I'm not completely positive if the chromeOS bug is tied to what we were experiencing on raw linux, but I suspect it is. Might be worth looking at what mesa version you're using in your crostini container! 

Here is the comment where we figured it out for reference:
https://github.com/zed-industries/zed/issues/14101#issuecomment-2254601539

**Comment by Trombach** [comment]
> Hey, I'm not completely positive if the chromeOS bug is tied to what we were experiencing on raw linux, but I suspect it is. Might be worth looking at what mesa version you're using in your crostini container!
> 
> Here is the comment where we figured it out for reference: [#14101 (comment)](https://github.com/zed-industries/zed/issues/14101#issuecomment-2254601539)

I'm also not sure if it's the same bug. For me this is happening on an AMD GPU and not an intel one. I've checked my mesa version and it's `22.3.6`, so much older than the problematic version mentioned in the other issue.

**Comment by darkoverlordofdata** [comment]
brand new Acer chromebook plus:

![Screenshot 2024-12-22 2 16 14 PM](https://github.com/user-attachments/assets/d5dabfdc-6be8-481a-b112-ad822820995e)

**Comment by darkoverlordofdata** [comment]
yeah the flicker went away but it still says it's unsupported if you aren't interested in supporting Chromebook why don't you just say so instead of playing bs games

![Image](https://github.com/user-attachments/assets/d0feeec1-b90f-492b-9806-d7e0a4865953)
